### PR TITLE
[clang-tidy] use emplace_back

### PR DESCRIPTION
**Pull request by neheb**
Found with modernize-use-emplace

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by arvidn** [comment]
```
'argument': conversion from '_Ty' to 'uint8_t', possible loss of data
        with
        [
            _Ty=int
        ]
    call "..\bin\standalone\msvc\17b09cf932770549210a83141190f492\msvc-setup.bat"  >nul
 cl /Zm800 -nologo @"..\bin\de4cc3570b8122ae43418614795d7a78\src\bdecode.obj.rsp" 
```

instantiated from here: `..\src\bdecode.cpp(882): note: see reference to function template instantiation`

**Comment by arvidn** [comment]
https://ci.appveyor.com/project/arvidn/libtorrent/builds/33496100/job/megth2eqgovpinr3#L265

**Comment by neheb** [comment]
No real idea how to fix.

File: src/bdecode.cpp [hunk 1]
```diff
@@ -878,8 +878,8 @@ namespace aux {
 						start = int_start;
 						TORRENT_FAIL_BDECODE(e);
 					}
-					ret.m_tokens.push_back({int_start - orig_start
-						, 1, bdecode_token::integer, 1});
+					ret.m_tokens.push_back(int_start - orig_start
```
Comment (arvidn) [inline src/bdecode.cpp:881]: I think you meant to say `emplace_back()` here.

**Comment by arvidn** [comment]
if you were to add `modernize-use-emplace` to the `.clang-tyidy` file, would it find more places. or would it pass CI?

**Comment by neheb** [comment]
It would. I intentionally left out one file since it doesn't compile with MSVC for some reason.

**Pull request closed without merging**